nexus/src/app/sagas/mod.rs

```diff
@@ -70,6 +72,17 @@ pub(crate) trait NexusSaga {
         params: &Self::Params,
         builder: steno::DagBuilder,
     ) -> Result<steno::Dag, SagaInitError>;
+
+    fn prepare(
+        params: &Self::Params,
+    ) -> Result<SagaDag, omicron_common::api::external::Error> {
+        let builder = DagBuilder::new(SagaName::new(Self::NAME));
+        let dag = Self::make_saga_dag(&params, builder)?;
+        let params = serde_json::to_value(&params).map_err(|e| {
+            SagaInitError::SerializeError(String::from("saga params"), e)
```

Comment:
take it or leave it: do you think it would be helpful if we actually included the params here?
```suggestion
            SagaInitError::SerializeError(format!("saga params: {params:?}"), e)
```
in practice, a serialization failure here seems pretty unlikely, so i can imagine wanting to know what the params we were trying to serialize were...

Reply:
Sure I can do that.  I'm a little on the fence because it's pretty incomplete (there are a ton of other users of `SerializeError` that follow the same convention as here, which is that they're just reporting what was being serialized).  I'm tempted to have `SerializeError` store a second `String` and make a constructor that accepts an `impl Debug` and generates a variant that has the name of the thing ("saga params"), the debug output of the thing we tried to serialize, and the error (like we have today).  That's a bigger change I don't want to take on right now.   Changing this one call site would help at least this one case if it comes up.